### PR TITLE
Issue #1287 - Corrected error message after 10 failed login attempts

### LINKED ISSUE #1287: Incorrect error message after multiple failed login attempts
Platform: Desktop and Mobile

Environment: Staging

Description of issue: When inputting incorrect credentials in the login page: https://fightpandemics.com/auth/login, with multiple failed attempted, an error message showed the following: "Login failed, reason: Internal Server Error". This can cause confusion if the credentials are incorrect or the frontend is not able to communicate with the backend.

Replicate issue:
1. Visit the main page: https://staging.fightpandemics.work/
2. Click the "Sign In" button
3. Input incorrect credentials in the following field:
Email: test@test.com
Password: TestTest1
4. Click the "Sign In" button 10 times.
5. Observe the incorrect error message.

Solution:
- Change the error message on the client side to reflect the rate limiting configured on Nginx.

### PR DESCRIPTION
**Pull request by jeremylin8**
### What does this pull request aim to achieve? 💯

When inputting incorrect credentials in the login page: https://fightpandemics.com/auth/login, with multiple failed attempted, an error message should show a user that the maximum Login attempts exceeded.
closes #1287 

_Please be concise and link any related issue(s):_

- [ x] I have checked that no one else is working on similar changes.
- [ x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x ] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix. The branch is created in a cloned version of this repo, **not a forked repo**.
- [ x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x ] There are no merge conflicts.
- [x ] There are no console warnings and errors.
- [x ] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

**Review by mannykary — APPROVED** [review]
~LGTM~; also tested in the review build by following the repro steps in the ticket and the correct error message shows up as expected.

EDIT: Actually, taking back my approval, sorry! I noticed something and made a [comment](https://github.com/FightPandemics/FightPandemics/pull/1318#discussion_r464764143).

File: backend/lib/endpoints/auth.js [hunk 1]
```diff
@@ -137,7 +137,9 @@ async function routes(app) {
         throw app.httpErrors.unauthorized("Wrong email or password.");
       }
       req.log.error(err, "Error logging in");
-      throw app.httpErrors.internalServerError();
+      throw app.httpErrors.internalServerError(
```
Comment (mannykary) [inline backend/lib/endpoints/auth.js]: Actually, looking closer, would we be able to parse the `err` object to see if the error is indeed due to too many login attempts? I can foresee some other errors being returned that may not necessarily be related to too many logins.

**Review by mannykary — CHANGES_REQUESTED** [review]

**Review by idjevm — CHANGES_REQUESTED** [review]
LGTM, except that we should change the message to `Maximum number of sign in attempts exceeded.`. The reason being is that we changed from Login to Sign In not long ago.

File: backend/lib/endpoints/auth.js (hunk 1 above)
Comment (idjevm) [reply backend/lib/endpoints/auth.js]: that's also true, according to the documentation of Nginx, the server will throw a 503 error, however, that can happen with other reasons too. I don't think the error message will say anything about the rate-limiting though, but rather just "service unavailable".

Comment (mannykary) [reply backend/lib/endpoints/auth.js]: I think Auth0 returns 429 for too many requests: https://auth0.com/docs/policies/rate-limits. Also, I see this error in [Papertrail](https://my.papertrailapp.com/groups/20211842/events?focus=1227132625207083073&q=program%3Afp-backend-feature1287&selected=1227132625207083073): 

image

Seems like we could either check the status code for `429` or check the error message for `TooManyRequestsError`.

Comment (jeremylin8) [reply backend/lib/endpoints/auth.js]: Yes, I agree. We should handle 429 status code error separately

Comment (idjevm) [reply backend/lib/endpoints/auth.js]: Awesome! then 429 it is :)

**Review by idjevm — APPROVED** [review]
LGTM. I will later open an issue to reword from `login` to `sign in`. This is great, thank you @jeremylin8 and @mannykary !!

**Comment by jeremylin8** [comment]
Sounds great. Thanks @idjevm and @mannykary :)

**Review by mannykary — APPROVED** [review]
LGTM

**Pull request merged**